apps/studio/data/analytics/warehouse-access-tokens-create-mutation.ts

```diff
@@ -0,0 +1,61 @@
+import { UseMutationOptions, useMutation, useQueryClient } from '@tanstack/react-query'
+import { analyticsKeys } from './keys'
+import { handleError, post } from 'data/fetchers'
+import { ResponseError } from 'types'
+import toast from 'react-hot-toast'
+
+type WarehouseAccessTokenCreateVariables = {
+  ref: string
+  description: string
+}
+async function createWarehouseAccessToken({
+  ref,
+  description,
+}: WarehouseAccessTokenCreateVariables) {
+  const { data, error } = await post(`/platform/projects/{ref}/analytics/warehouse/access-tokens`, {
+    params: {
+      path: { ref },
+    },
+    body: { description },
+  } as any) // TODO: remove type cast when fetcher fn params are typed
+
+  if (error) handleError(error)
+
+  return data
+}
+
+type WarehouseAccessTokenUpdateData = Awaited<ReturnType<typeof createWarehouseAccessToken>>
+
+export function useCreateWarehouseAccessToken({
+  onSuccess,
+  onError,
+  ...options
+}: Omit<
+  UseMutationOptions<
+    WarehouseAccessTokenUpdateData,
+    ResponseError,
+    WarehouseAccessTokenCreateVariables
+  >,
+  'mutationFn'
+> = {}) {
+  const queryClient = useQueryClient()
+
+  return useMutation(
+    (payload: WarehouseAccessTokenCreateVariables) => createWarehouseAccessToken(payload),
+    {
+      async onSuccess(data, vars, context) {
+        const keysToInvalidate = analyticsKeys.warehouseAccessTokens(vars.ref)
+        await queryClient.invalidateQueries(keysToInvalidate)
```

Comment:
for consistency, can just be
```suggestion
        await queryClient.invalidateQueries(analyticsKeys.warehouseAccessTokens(vars.ref))
```